src/lib/instrument-listen.js

```diff
@@ -11,24 +11,20 @@ const readFile = denodeify(fs.readFile);
 
 module.exports = (app, meta, initPromises) => {
 	async function createServer(app) {
-		if (process.argv.indexOf('--https') > -1) {
+		if (process.argv.includes('--https')) {
 			const [key, cert] = await Promise.all([
-				readFile(path.resolve(__dirname, '../../key.pem')),
-				readFile(path.resolve(__dirname, '../../cert.pem'))
-			])
+				readFile(path.resolve(process.cwd(), 'self-signed-ssl-key.pem')),
+				readFile(path.resolve(process.cwd(), 'self-signed-ssl-certificate.pem'))
+			]).catch(() => {
+				throw Error('n-express was started with --https, but there\'s no self-signed certificate or key in your app directory. run `npx n-express-generate-certificate` to create one')
+			});
 
 			return https.createServer({ key, cert }, app);
 		} else {
 			return http.createServer(app);
 		}
 	}
 
-	async function actualAppListen(app, ...args) {
```

Comment:
trying to understand why this `actualAppListen` function was being used in the first place, it looks that what you changed it for is doing the same thing, but I am wondering if I am missing something.

Reply:
ahh I had not seen that the function was being assigned to `app.listen` on line 32, thanks!